webodf/lib/gui/ZoomHelper.js

```diff
@@ -443,13 +443,31 @@
         };
 
         /**
-         * @param {!number} zoomLevel
-         * @return {undefined}
-         */
+        * Sets the zoom level of the document and maintains the top left position
+        * @param {!number} zoomLevel
+        * @return {undefined}
+        */
         this.setZoomLevel = function (zoomLevel) {
             if (zoomableElement) {
+                var scrollRatios = [];
+                var currentOffsetParent = offsetParent;
+                // Save the scroll ratios and related parent
+                while (currentOffsetParent) {
+                    scrollRatios.push({ container: currentOffsetParent, scrollTop: currentOffsetParent.scrollTop / zoom,
```

Comment:
Minor, would generally have a line for each new container property when they don't all fit on the same line. And would do just a single indent. E.g.:
```
scrollRatios.push({
  container: currentOffsetParent, 
  scrollTop: currentOffsetParent.scrollTop / zoom,
  scrollLeft: currentOffsetParent.scrollLeft / zoom
});
```